src/plugins/data/invocation/Resource.ts

```diff
@@ -50,6 +53,9 @@ class Resource {
                 });
             }
             this.resourceDuration += resource.resourceDuration;
+            if (resource.resourceDuration > this.resourceMaxDuration) {
```

Comment:
I think this should be `resource.resourceMaxDuration > this.resourceMaxDuration` as we are merging two resources